main.scss

```
@@ -12,14 +20,154 @@
 ///		))
 @mixin oAutocomplete ($opts: ()) {
 	// content of primary mixin
-	.o-autocomplete {
+	.o-autocomplete .o-autocomplete__container {
+		display: inline-grid;
+		position: relative;
+	}
+
+	.o-autocomplete .o-autocomplete__container > div:first-child {
+		@include oTypographySans($scale: 0);
+		grid-row: 1;
+		grid-column: 1;
+	}
```

I think there's an opportunity to reduce some specificity here, by not applying layout styles to alphagov elements. 

```suggestion
	.o-autocomplete {
		display: inline-grid;
		position: relative;
	}

	.o-autocomplete__container {
		@include oTypographySans($scale: 0);
		grid-row: 1;
		grid-column: 1;
	}
```

For the above to work the `addOrRemoveClearButton` function would also need to be updated to append the clear button to the root o-autocomplete element rather than the alphagov container. 

With those changes the close button will be miss-aligned on the loading state. To fix that we'd need to make the loading state absolutely positioned in the same way as the result list. Which will also avoid pushing page content down when loading (see how the lorem text is shifted down on loading in the example below).

https://user-images.githubusercontent.com/10405691/122070487-eb169100-cded-11eb-876b-e9942e84c786.mp4

---

Another option would be to remove the container, styling `.o-autocomplete__wrapper` instead. Then the container/wrapper nonsense naming is avoided but there's a greater chance alphagov changes will conflict with o-autocomplete

---

> Another option would be to do the above but remove the container, styling `.o-autocomplete__wrapper` instead. Then the container/wrapper nonsense naming is avoided but there's a greater chance alphagov changes will conflict with o-autocomplete in the future

We can avoid the potential conflict issue by pinning to a specific version of accessibile-autocomplete and updating to newer versions manually. That way Percy would catch any conflicts for us.

Which approach would you recommend going with?

---

👍 let's remove the container and pin

---

As requested, I had a go at this :) https://github.com/Financial-Times/o-autocomplete/pull/3